python/api/cluster/serializers.py

```diff
@@ -94,6 +94,32 @@ def update(self, instance, validated_data):
         return instance
 
 
+class ClusterUISerializer(ClusterSerializer):
+    action = CommonAPIURL(view_name='object-action')
+    edition = CharField(read_only=True)
+    prototype_version = SerializerMethodField()
+    prototype_name = SerializerMethodField()
+    prototype_display_name = SerializerMethodField()
+    upgrade = hlink('cluster-upgrade', 'id', 'cluster_id')
```

Comment:
`hlink` function seems to be redundant and add implicitness

Reply:
fixed